udp_driver/include/udp_manager_node.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef UDP_DRIVER__UDP_MANAGER_NODE_HPP_
+#define UDP_DRIVER__UDP_MANAGER_NODE_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <map>
+#include <memory>
+
+#include "udp_msgs/msg/udp_packet.hpp"
+#include "udp_msgs/srv/udp_send.hpp"
+#include "udp_msgs/srv/udp_socket.hpp"
+
+namespace udp_driver
+{
+
+class UdpManagerNode : public rclcpp::Node
+{
+  public:
+    UdpManagerNode();
+    ~UdpManagerNode(){}
+
+  private:
+    rclcpp::Service<udp_msgs::srv::UdpSocket>::SharedPtr create_udp_driver_;
+    std::map<uint16_t, std::shared_ptr<udp_msgs::srv::UdpSocket>> active_udp_drivers_;
+
+    void create_driver_handler(
+      const std::shared_ptr<udp_msgs::srv::UdpSocket::Request> request,
+      const std::shared_ptr<udp_msgs::srv::UdpSocket::Response> response);
+};
+}  //  namespace udp_driver
+
+int main(int argc, char **argv)
+{
+  rclcpp::init(argc, argv);
+
+  std::shared_ptr<udp_driver::UdpManagerNode> node = std::make_shared<udp_driver::UdpManagerNode>();
+
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "UDP Manager node is up and running");
+  rclcpp::spin(node);
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "UDP Manager node is shutting down...");
+  rclcpp::shutdown();
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Exiting...");
+  return 0;
```

Comment:
From `dashing` forward, it is better to register this as a component. In addition to allowing this node to be grouped with others in a single process, it allows you to remove this boilerplate by using  the following in CMakeLists.txt:

```cmake
rclcpp_components_register_node(${PROJECT_NAME}_node
  PLUGIN "udp_driver::UdpManagerNode"
  EXECUTABLE ${PROJECT_NAME}_node_exe
)
```

(or similar)

This will generate the executable with the `main` function for you.

See https://autowarefoundation.gitlab.io/autoware.auto/AutowareAuto/contributor-guidelines.html#autotoc_md21 for a quick overview of the process.

Reply:
You registered the node and set it to build the executable but I think you forgot to remoe `main` here.